server/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -131,7 +120,10 @@ public MatchQueryBuilder(StreamInput in) throws IOException {
         minimumShouldMatch = in.readOptionalString();
         fuzzyRewrite = in.readOptionalString();
         fuzziness = in.readOptionalWriteable(Fuzziness::new);
-        cutoffFrequency = in.readOptionalFloat();
+        // cutoff_frequency has been removed
+        if (in.getVersion().onOrBefore(Version.V_8_0_0)) {
```

Comment:
It should be strictly `before` 8.0 so that we can remove in 9.

Reply:
Of course, thank you!